specs/networking/p2p-interface.md

```diff
@@ -0,0 +1,743 @@
+
+## Multiplexing
+
+During connection bootstrapping, libp2p dynamically negotiates a mutually supported multiplexing method to conduct parallel conversations. This applies to transports that are natively incapable of multiplexing (e.g. TCP, WebSockets, WebRTC), and is omitted for capable transports (e.g. QUIC).
+
+Two multiplexers are commonplace in libp2p implementations: [mplex](https://github.com/libp2p/specs/tree/master/mplex) and [yamux](https://github.com/hashicorp/yamux/blob/master/spec.md). Their protocol IDs are, respectively: `/mplex/6.7.0` and `/yamux/1.0.0`.
+
+Clients MUST support [mplex](https://github.com/libp2p/specs/tree/master/mplex) and MAY support [yamux](https://github.com/hashicorp/yamux/blob/master/spec.md). If both are supported by the client, yamux must take precedence during negotiation. See the Rationale section of this document for tradeoffs.
+
+# ETH2 network interaction domains
+
+## Constants
+
+This section outlines constants that are used in this spec.
+
+- `RQRP_MAX_SIZE`: The max size of uncompressed req/resp messages that clients will allow.
+ Value: TBD
+- `GOSSIP_MAX_SIZE`: The max size of uncompressed gossip messages
+ Value: 1MB (estimated from expected largest uncompressed block size).
+- `SHARD_SUBNET_COUNT`: The number of shard subnets used in the gossipsub protocol.
+ Value: TBD
+
+## The gossip domain: gossipsub
+
+Clients MUST support the [gossipsub](https://github.com/libp2p/specs/tree/master/pubsub/gossipsub) libp2p protocol.
+
+**Protocol ID:** `/meshsub/1.0.0`
+
+**Gossipsub Parameters**
+
+*Note: Parameters listed here are subject to a large-scale network feasibility study.*
+
+The following gossipsub parameters will be used:
+
+- `D` (topic stable mesh target count): 6
+- `D_low` (topic stable mesh low watermark): 4
+- `D_high` (topic stable mesh high watermark): 12
+- `D_lazy` (gossip target): 6
+- `fanout_ttl` (ttl for fanout maps for topics we are not subscribed to but have published to, seconds): 60
+- `gossip_advertise` (number of windows to gossip about): 3
+- `gossip_history` (number of heartbeat intervals to retain message IDs): 5
+- `heartbeat_interval` (frequency of heartbeat, seconds): 1
+
+### Topics
+
+Topics are plain UTF-8 strings, and are encoded on the wire as determined by protobuf (gossipsub messages are enveloped in protobuf messages).
+
+Topic strings have form: `/eth2/TopicName/TopicEncoding`. This defines both the type of data being sent on the topic and how the data field of the message is encoded. (Further details can be found in [Messages](#Messages)).
+
+There are two main topics used to propagate attestations and beacon blocks to all nodes on the network. Their `TopicName`'s are:
+
+- `beacon_block` - This topic is used solely for propagating new beacon blocks to all nodes on the networks. Blocks are sent in their entirety. Clients who receive a block on this topic MUST validate the block proposer signature before forwarding it across the network.
```

Comment:
we probably should also validate the block is not older than certain slot (i.e. current finalized slot)

Reply:
Yes, adding such a validation rule reduces DoS vectors.

Reply:
that is already implied - once finalization, all blocks not on the canonical chain before it become invalid by definition.